contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -30,13 +31,28 @@ contract MixinStakingPoolRewards is
 {
     using LibSafeMath for uint256;
 
+    /// @dev Syncs the rewards for the sender and then withdraws all of the eth vault to the sender.
+    /// @param poolIds The ids of the pools that should be withdrawn from.
+    function withdrawFromPools(bytes32[] calldata poolIds)
+        external
+        returns (uint256)
+    {
+        uint256 length = poolIds.length;
+
+        for (uint256 i = 0; i != length; i++) {
```

Comment:
So this function deposits the delegator's ETH into the `ethVault`, does some accounting, and then withdraws the delegator's ETH from the `ethVault`? Why transfer to the  `ethVault` at all? The withdrawal patter also forces the `ethVault` to trust the `StakingProxy`. 

I'm ~95% certain we can completely remove the use of the `ethVault` for delegators. I'll dig into it more tomorrow.

Reply:
Here's a quick implementation. I'm guessing we can remove more complexity if we follow along these lines: https://github.com/0xProject/0x-monorepo/compare/3.0...feat/3.0/direct-withdrawals?expand=1

We should also rename the vaults to `OperatorRewardVault` and `MembersRewardVault`.